effekt/shared/src/main/scala/effekt/Namer.scala

```diff
@@ -571,9 +571,13 @@ object Namer extends Phase[Parsed, NameResolved] {
       case _ => Context.abort(s"Can only assign to mutable variables.")
     }
 
-    case tpe: source.ValueType    => resolve(tpe)
     case tpe: source.FunctionType => resolve(tpe)
 
+    // NOTE(jiribenes, 2025-04-21):
+    //   How do we get around this case? How do we know which one is expected? Do we even need to do this?
+    //   ... or should we just choose based on the syntax?
+    case tpe: source.Type         => Context.abort(s"Unexpected kindless type: $tpe.")
```

Comment:
Is this error user facing? Kindless type is not something we have at the moment as a concept.

Reply:
I wasn't able to trigger it in any way...

Reply:
We can somewhat easily ("just implementation effort") get rid of all of this by defining the `resolve` functions manually for all kinds of trees (`Stmt`, `Term`, `Def`, `Implementation`, etc.).

In general, I think this is a good thing to do anyways to simplify the implementation. However, this will massively clash with other ongoing refactorings, so I don't do this now.

Reply:
Then this case does not show up by construction, since `Type` is not a `Term` and we use `resolveValueType` elsewhere.